### PR TITLE
Souryadeep

### PR DESCRIPTION
**Pull request by Souryadeep0325**

### CODE REVIEW

File: Grocery App/AddCategoryViewController.swift [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import FirebaseCore
+import FirebaseFirestore
+protocol PassAction {
+    func addTapped(Name: String)
+}
+class AddCategoryViewController: UIViewController, UITextFieldDelegate {
+
+    @IBOutlet weak var newCategoryTxtField: UITextField!
+    @IBOutlet weak var newRankTxtField: UITextField!
+    var uid = "nil"
```
Comment (KrishanDew) [inline Grocery App/AddCategoryViewController.swift]: #12

File: Grocery App/AddCategoryViewController.swift [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        }
+        return s
+    }
+    
+    func setData() {
+        let db = Firestore.firestore()
+        let id = randomString(of: 20)
+        db.collection("categories").document("\(id)").setData([
+            "id": "\(id)",
+            "name": "\(newCategoryTxtField.text ?? "nil")",
```
Comment (KrishanDew) [inline Grocery App/AddCategoryViewController.swift]: #12

File: Grocery App/AddCategoryViewController.swift [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    func addTapped(Name: String)
+}
+class AddCategoryViewController: UIViewController, UITextFieldDelegate {
+
+    @IBOutlet weak var newCategoryTxtField: UITextField!
+    @IBOutlet weak var newRankTxtField: UITextField!
+    var uid = "nil"
+    var category = " "
+    var rank = " "
+    var selectionDelegate: PassAction!
```
Comment (KrishanDew) [inline Grocery App/AddCategoryViewController.swift]: Delegate property must be optional not forcefully unwrapped.

File: Grocery App/AddCategoryViewController.swift [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            
+        }
+        // Do any additional setup after loading the view.
+    }
+    
+
+    @IBAction func addButtonTapped(_ sender: Any) {
+        if uid == "nil" {
+        setData()
+        selectionDelegate?.addTapped(Name: "yes")
```
Comment (KrishanDew) [inline Grocery App/AddCategoryViewController.swift]: parameter name must be start with small character

File: Grocery App/AddCategoryViewController.swift [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        }
+        else {
+            let db = Firestore.firestore()
+            
+            db.collection("categories").document("\(uid)").updateData([
+                "name": "\(newCategoryTxtField.text ?? "nil")",
```
Comment (KrishanDew) [inline Grocery App/AddCategoryViewController.swift]: #12

File: Grocery App/AddProductViewController.swift [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+//  Created by Souryadeep Sadhukhan on 09/05/21.
+//
+
+import UIKit
+import FirebaseAuth
+import FirebaseCore
+import FirebaseFirestore
+import DropDown
+
+class CellClass1: UITableViewCell {
```
Comment (KrishanDew) [inline Grocery App/AddProductViewController.swift]: Class name must be proper name.

File: Grocery App/AddProductViewController.swift [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+    var dataStorage = [[String]]()
+    @IBOutlet weak var activeDropDownButton: UIButton!
+    @IBOutlet weak var descriptionDropDownButton: UIButton!
+    var productCategoryDict = [[String]]()
+    @IBOutlet weak var productName: UITextField!
+    @IBOutlet weak var productDescription: UITextField!
+    
+    @IBOutlet weak var productPrice: UITextField!
+    
+    @IBOutlet weak var ProductCategory: UITextField!
```
Comment (KrishanDew) [inline Grocery App/AddProductViewController.swift]: must start with small character.

Comment (Souryadeep0325) [reply Grocery App/AddProductViewController.swift]: done

File: Grocery App/AddProductViewController.swift (hunk 6 above)
Comment (Souryadeep0325) [reply Grocery App/AddProductViewController.swift]: done

File: Grocery App/AddCategoryViewController.swift (hunk 5 above)
Comment (Souryadeep0325) [reply Grocery App/AddCategoryViewController.swift]: Done sir

File: Grocery App/AddCategoryViewController.swift (hunk 4 above)
Comment (Souryadeep0325) [reply Grocery App/AddCategoryViewController.swift]: Done sir

File: Grocery App/AddCategoryViewController.swift (hunk 3 above)
Comment (Souryadeep0325) [reply Grocery App/AddCategoryViewController.swift]: Done Sir

File: Grocery App/AddCategoryViewController.swift (hunk 2 above)
Comment (Souryadeep0325) [reply Grocery App/AddCategoryViewController.swift]: Done sir

File: Grocery App/AddCategoryViewController.swift (hunk 1 above)
Comment (Souryadeep0325) [reply Grocery App/AddCategoryViewController.swift]: Done Sir

**Pull request closed without merging**